### ISSUE TITLE
Support Ruby 2.5+

### ISSUE THREAD

**Issue by jodosha** [hanami/hanami#921]
Ruby is moving faster than us. 😉 💎 

---

For **all** the Hanami gems:

  - [x] Change the minimum supported Ruby version in `.gemspec` to `">= 2.5.0"`
  - [x] Remove `2.3`, and `2.4` from `.travis.yml` and `.circleci/.config.yml`
  - [x] Remove `jruby-9.1.0.0` from `.travis.yml` and `.circleci/.config.yml`
  - [x] Update `README` with supported Ruby version

---

Changes **MUST** go to `unstable` branches of each repository

**Comment by jodosha** [comment]
Folks, we have to put this on hold.

We can't jump already into 2.4 because of JRuby.
It will support Ruby 2.4 as of 9.2.0.0.
If we bump Rubocop's `TargetRubyVersion`, it will make our builds to fail, due to the 2.4 specific suggestions.

On the other hand, if we bump already the min version in our gems 2.4,
the builds will still fail because JRuby 9.1 corresponds to MRI 2.3.

So we're locked for now. 😢

**Comment by pchaganti** [comment]
Looks like JRuby 9.2 is getting closer :)

**Comment by jodosha** [comment]
🤞

**Comment by 007lva** [comment]
JRuby 9.2 is here!
http://jruby.org/2018/05/24/jruby-9-2-0-0.html

**Comment by cllns** [comment]
Thanks @007lva :D 

It looks like they actually support Ruby 2.5 as well!

@hanami/core I wonder if we should jump to support Ruby 2.5 for Hanami 2.0, and skip 2.4? What do you think?

**Comment by jodosha** [comment]
@cllns My opinion is to keep 2.4+ for now. If 2.0 will be released in the far future, then yes, we can consider 2.5+.

**Comment by jodosha** [comment]
This is still blocked by Travis CI that apparently doesn't have JRuby 9.2.0.0 yet.

**Comment by jodosha** [comment]
Please hold on with this. We need first to reach JRuby 9.2 compat for JRuby with `hanami-model` and `hanami-view` and then we'll be able to proceed.

**Comment by waghanza** [comment]
@jodosha Taking about https://github.com/hanami/hanami/issues/921#issuecomment-395693994, requiring  `>=2.5` for **hanami** `2.0` is going to restraint **hanami** adoption, because versio of `ruby` on `major` **linux** :

+ amazonlinux => `2.4`
+ debian => `2.3`
+ ubuntu => `2.5`
+ centos => 2.4 (via `software collection`)

PS : Those are LTS

**Comment by graywolf** [comment]
Can't you always just install new version using rvm? Or from backports?

**Comment by waghanza** [comment]
@graywolf I surely can do this 😁 however using `rvm` or **backport** packages (you are assuming I'm on debian-based) add unecessary performance decrease 

I can also compile `ruby` 😉 but my comment was more a reaction than a complain

**Comment by jodosha** [comment]
@waghanza It's a small friction, true. I expect that Ruby apps are run not via a OS Ruby, but from official MRI, perhaps via a version manager (eg. chruby, rbenv, rvm..).

**Comment by waghanza** [comment]
I can understand your argument here @jodosha, but
+ Any versionning ssystem add unecessary syscallls
+ Using system  Ruby ensure stability, monkey patching ...

**Comment by deepj** [comment]
Honestly, I don't know anyone who uses system Ruby for running their application. Especially for web applications.

**Comment by waghanza** [comment]
@deepj me :stuck_out_tongue: 
I prefer using **binary** that has been `compiled` before (an not by me) to ensure **stability**, because of lib version ...

btw, it could be done with `rvm` but I think it add flexibility to the detriment of performances

**Comment by 007lva** [comment]
[Rails 6 wil require Ruby version >= 2.5](https://github.com/rails/rails/pull/32034).

Ubuntu 18.04 and Debin 10 ships Ruby 2.5.1

I think that now sounds more reasonable to require Ruby 2.5 for Hanami 2.0

**Comment by jodosha** [comment]
@007lva agreed
